src/Core/Impl/Extensions/IOExtensions.cs

```diff
@@ -114,28 +114,28 @@ public static class IOExtensions {
             for (var retries = 100; retries > 0; --retries) {
                 try {
                     return fs.ReadAllText(file);
-                } catch (UnauthorizedAccessException) {
-                    Thread.Sleep(10);
-                } catch (IOException) {
+                } catch (Exception ex) when (ex is UnauthorizedAccessException || ex is IOException || ex is ObjectDisposedException) {
                     Thread.Sleep(10);
                 }
             }
             return null;
         }
 
         public static void WriteTextWithRetry(this IFileSystem fs, string filePath, string text) {
-            try {
-                using (var stream = fs.OpenWithRetry(filePath, FileMode.Create, FileAccess.Write, FileShare.Read)) {
-                    if (stream != null) {
-                        var bytes = Encoding.UTF8.GetBytes(text);
-                        stream.Write(bytes, 0, bytes.Length);
-                        return;
-                    }
+            var failed = false;
+            for (var retries = 100; retries > 0; --retries) {
+                try {
+                    fs.WriteAllText(filePath, text);
+                } catch (Exception ex) when (ex is IOException || ex is UnauthorizedAccessException) {
+                    failed = true;
```

Comment:
If there was one exception and next retry succeeds, file will be deleted later. We need either return after `fs.WriteAllText` or set `failed` to `true`.